example/bench/CMakeLists.txt

```diff
@@ -2,3 +2,4 @@ add_executable(bench EXCLUDE_FROM_ALL bench.cpp)
 add_dependencies(examples bench)
 
 target_link_libraries(bench PRIVATE arbor arborenv arbor-sup ${json_library_name})
+set_target_properties(bench PROPERTIES CXX_VISIBILITY_PRESET hidden)
```

Comment:
I removed the hidden visibility again. Now, I could not get rid of some warnings on macos when static libraries are built, however, the tests pass. One way to avoid the warnings would be to compile with default visibility in the special case of macos+static. If the warnings are not bothering anyone, we could go with what we have.